.github/workflows/build-test-distribute.yaml

```diff
@@ -170,3 +173,153 @@ jobs:
+
+          RUN_FULL_MATRIX="${{ contains(github.event.pull_request.labels.*.name, 'ci/run-full-matrix') }}"
+          if [[ "$RUN_FULL_MATRIX" == "false" ]]; then
+            echo "Skipping non priority e2e tests, because no label 'ci/run-full-matrix' was found on the pull request."
+
+            SLOW=''         # target==""
+            DELTA_KDS=''    # deltaKDS=="true"
+            CNI=''          # cniNetworkPlugin=="calico"
+            MAIN=$(echo $MAIN | $JQ_PATH -rc ".exclude += $EXCLUDE_ARM64 | .exclude += $EXCLUDE_MAIN_NON_PRIORITY")
+          fi
+
+          echo "slow=$SLOW" >> $GITHUB_OUTPUT
+          echo "main=$MAIN" >> $GITHUB_OUTPUT
+          echo "delta_kds=$DELTA_KDS" >> $GITHUB_OUTPUT
+          echo "cni=$CNI" >> $GITHUB_OUTPUT
+  e2e_slow:
+    name: "legacy-k8s:${{ matrix.arch }}-${{ matrix.k8sVersion }}"
+    runs-on: ubuntu-latest
+    needs: ["gen_e2e_matrix"]
+    if: needs.gen_e2e_matrix.outputs.slow != ''
+    strategy:
+      matrix: ${{ fromJSON(needs.gen_e2e_matrix.outputs.slow) }}
+    steps:
+      - run: |
+          echo "k8sVersion: ${{ matrix.k8sVersion }}"
+          echo "arch: ${{ matrix.arch }}"
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: ./.github/actions/e2e
+        with:
+          k8sVersion: ${{ matrix.k8sVersion }}
+          arch: ${{ matrix.arch }}
+          parallelism: 3
+          target: ""
+  e2e_main:
+    name: "${{ matrix.target }}:${{ matrix.arch }}-${{ matrix.k8sVersion }}"
+    runs-on: ubuntu-latest
+    needs: ["gen_e2e_matrix"]
+    if: needs.gen_e2e_matrix.outputs.main != ''
+    strategy:
+      matrix: ${{ fromJSON(needs.gen_e2e_matrix.outputs.main) }}
+    steps:
+      - run: |
+          echo "k8sVersion: ${{ matrix.k8sVersion }}"
+          echo "target: ${{ matrix.target }}"
+          echo "arch: ${{ matrix.arch }}"
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: ./.github/actions/e2e
+        with:
+          k8sVersion: ${{ matrix.k8sVersion }}
+          target: ${{ matrix.target }}
+          arch: ${{ matrix.arch }}
+  e2e_delta_kds:
+    name: "${{ matrix.target }}:${{ matrix.arch }}-${{ matrix.k8sVersion }}-delta-kds"
+    runs-on: ubuntu-latest
+    needs: ["gen_e2e_matrix"]
+    if: needs.gen_e2e_matrix.outputs.delta_kds != ''
+    strategy:
+      matrix: ${{ fromJSON(needs.gen_e2e_matrix.outputs.delta_kds) }}
+    steps:
+      - run: |
+          echo "k8sVersion: ${{ matrix.k8sVersion }}"
+          echo "target: ${{ matrix.target }}"
+          echo "arch: ${{ matrix.arch }}"
+          echo "deltaKDS: ${{ matrix.arch }}"
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: ./.github/actions/e2e
+        with:
+          k8sVersion: ${{ matrix.k8sVersion }}
+          target: ${{ matrix.arch }}
+          arch: ${{ matrix.arch }}
+          deltaKDS: ${{ matrix.deltaKDS }}
+  e2e_calico:
```

Comment:
as mentioned let's remove the delta job.

Reply:
See the previous conversation.